design/thin-edge-core.md

```diff
@@ -0,0 +1,203 @@
+# Thin-edge Core Vision
```

Comment:
@didier-wenzek  I propose to rename this whole document to thin-edge.io Technology Vsion & Strategy or Design Principles and to link it the the vision.md:
1. In the vision md explains our motivation, target group/users and their needs, high level summary of the framework capabilities 
2. this document should explain how we get there in terms of technology choices , design principles etc . The audience of this doc should be anyone who is willing to user, extend or contribute to thin-edge.io